Examples/MAX78002/SPI/main.c

```diff
@@ -143,48 +133,53 @@ int main(void)
 
     for (i = 2; i < 17; i++) {
         // Sending out 2 to 16 bits
-
         for (j = 0; j < DATA_LEN; j++) {
             tx_data[j] = DATA_VALUE;
         }
 
-        // Configure the peripheral
-        retVal = MXC_SPI_Init(SPI, 1, 0, 1, 0, SPI_SPEED, spi_pins);
+        init.spi = SPI;
+        init.freq = SPI_SPEED;
+        init.spi_pins = NULL; // Use default, predefined pins
+        init.mode = MXC_SPI_INTERFACE_STANDARD;
+        init.type = MXC_SPI_TYPE_CONTROLLER;
+        init.clk_mode = MXC_SPI_CLKMODE_0; // CPOL: 0, CPHA: 0
+        init.frame_size = i;
+        init.ts_control = MXC_SPI_TSCONTROL_HW_AUTO;
+        init.target.active_polarity = 0;
+        init.vssel = MXC_GPIO_VSSEL_VDDIO;
+        init.ts_mask = 0x01; // Use Target Select 0
+
+#if MASTERDMA
+        init.use_dma = true;
+        init.dma = MXC_DMA;
+#else
+        init.use_dma = false;
+#endif
+
+        retVal = MXC_SPI_Init_v2(&init);
         if (retVal != E_NO_ERROR) {
             printf("\nSPI INITIALIZATION ERROR\n");
             return retVal;
         }
 
         memset(rx_data, 0x0, DATA_LEN * sizeof(uint16_t));
 
-        //SPI Request
+        // SPI Request
         req.spi = SPI;
-        req.txData = (uint8_t *)tx_data;
-        req.rxData = (uint8_t *)rx_data;
-        req.txLen = DATA_LEN;
-        req.rxLen = DATA_LEN;
-        req.ssIdx = 0;
-        req.ssDeassert = 1;
-        req.txCnt = 0;
-        req.rxCnt = 0;
-        req.completeCB = (spi_complete_cb_t)SPI_Callback;
+        req.tx_buffer = (uint8_t *)tx_data;
+        req.rx_buffer = (uint8_t *)rx_data;
+        req.tx_len = DATA_LEN;
+        req.rx_len = DATA_LEN;
+        req.index = 0;
+        req.deassert = 1;
+        req.tx_cnt = 0;
+        req.rx_cnt = 0;
+        req.callback = (mxc_spi_callback_t)SPI_Callback;
```

Comment:
The mxc_spi_req_t req object supports both the old and new names. I added the new names to match the code style of SPI_RevA2 and remove the slave select terms.